ynput/OpenPype
# Houdini: test silent publish

## Changelog Description
This Test PR extends https://github.com/ynput/OpenPype/pull/5891 's functionality. 
This PR adds two modes of publishing 
1. `Silent publish` (which runs the publish logic independently on the publisher)
2. `Use Publisher UI` (which leave the publisher UI to handle the publish)

![image](https://github.com/ynput/OpenPype/assets/20871534/98cb6da5-3916-4893-9e62-dd1da528bee5)

## Additional Info
1. This PR is compatible with any instances that were made before this PR.
2. Silent publish will print its log to Houdini's console and artists will lose the ability to change any exposed settings in the publisher UI. 

## Demo
https://github.com/ynput/OpenPype/assets/20871534/e6061351-063b-49e1-8785-9c857e4e7309

## How silent publish works
It's a headless publish with a customized context


## Testing notes:
1. Enable self publish button in the Houdini addon settings
    ![image](https://github.com/ynput/OpenPype/assets/20871534/bd72ae95-ecc6-4292-ab7b-266cf967e76b)
2. Create new instance and use `AYON Publish` Button. 
3. Test the two modes `Silent publish` and Use Publisher UI`
4. Try connecting other instances to the input your previously created instance. (they should be included in your publish)
